instrumentation/github.com/aws/aws-lambda-go/otellambda/xrayconfig/xrayconfig.go

```diff
@@ -35,72 +34,45 @@ func xrayEventToCarrier([]byte) propagation.TextMapCarrier {
 	return propagation.HeaderCarrier{"X-Amzn-Trace-Id": []string{xrayTraceID}}
 }
 
-type asyncSafeFlusher struct {
-	tp *sdktrace.TracerProvider
-}
-
-func (f asyncSafeFlusher) ForceFlush(ctx context.Context) error {
-	// yield processor to attempt to ensure all spans have
-	// been consumed and are ready to be flushed
-	// - see https://github.com/open-telemetry/opentelemetry-go/issues/2080
-	// to be removed upon resolution of above issue
-	runtime.Gosched()
-
-	return f.tp.ForceFlush(ctx)
-}
-
-// tracerProviderAndFlusher returns a list of otellambda.Option(s) to
-// enable using a TracerProvider configured for AWS XRay via a collector
-// and an otellambda.Flusher to flush this TracerProvider.
-// tracerProviderAndFlusher is not exported because it should not be used
-// without the provided EventToCarrier function and XRay Propagator
-func tracerProviderAndFlusher() ([]otellambda.Option, error) {
-	ctx := context.Background()
-
-	// Do not need transport security in Lambda because collector
-	// runs locally in Lambda execution environment
+// NewTracerProvider returns a TracerProvider configured with an exporter,
+// ID generator, and lambda resource detector to send trace data to AWS X-Ray
+// via a Collector instance listening on localhost
+func NewTracerProvider(ctx context.Context) (*sdktrace.TracerProvider, error) {
 	exp, err := otlptracegrpc.New(ctx, otlptracegrpc.WithInsecure())
 	if err != nil {
-		return []otellambda.Option{}, err
+		errorLogger.Println("failed to create exporter: ", err)
```

Comment:
We probably don't need to log here.  Since we return the error, we expect the caller to handle it (and log it if they would like).  Same applies below and elsewhere.